controllers/core/serving_controller.go

```diff
@@ -213,7 +216,13 @@ func (r *ServingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 
 func (r *ServingReconciler) getServingRun(s *openfunction.Serving) core.ServingRun {
 	if s.Spec.Triggers.Http != nil {
-		return knative.NewServingRun(r.ctx, r.Client, r.Scheme, r.Log)
+		if s.Spec.Triggers.Http.Engine == nil || *s.Spec.Triggers.Http.Engine == openfunction.HttpRuntimeKnative {
+			return knative.NewServingRun(r.ctx, r.Client, r.Scheme, r.Log)
+		} else if *s.Spec.Triggers.Http.Engine == openfunction.HttpRuntimeKeda {
+			return kedahttp.NewServingRun(r.ctx, r.Client, r.Scheme, r.Log)
+		} else {
+			return nil
+		}
```

Comment:
When `knative` is the default, it can be changed to:

```golang
if engine == keda {
    // keda new serving run
} else {
    // knative new serving run
}
```